docs/docsite/rst/testing_modules.rst

```diff
@@ -0,0 +1,196 @@
+            password: "{{ aci_password }}"
+            validate_certs: '{{ aci_validate_certs | default(false) }}'
+            use_ssl: '{{ aci_use_ssl | default(true) }}'
+            use_proxy: '{{ aci_use_proxy | default(true) }}'
+            output_level: "{{ aci_output_level | default('info') }}"
+
+4. The next section deletes the tenant. This ensures that we don't have the root object configuration on our APIC. This is done to avoid idempotency issues later during the creation of other objects pertaining to our module. We verify the result of each task in the test file, which also checks for idempotency. If an object such as the tenant already exists before the test begins, these verification tests may fail.
+
+.. code-block:: yaml
+
+  - name: Remove the ansible_tenant
+    aci_tenant:
+      <<: *aci_info 
+      tenant: ansible_tenant
+      state: absent
+
+.. note::
+
+  - A cleanup section is included before we start testing to remove any existing objects, ensuring a clean state. 
+  - Another cleanup section is added after testing to delete created objects, since not all objects are explicitly defined within the tenant.
+
+The test cases for the ACI modules follow a consistent pattern across operations such as Create, Update, Query, Query all, and Delete. For Create, Update, and Delete operations, each test includes:
+
+  + The check mode task to simulate the operation without making changes.
+  + The normal_mode task to perform the actual operation.
+  + The normal mode tasks to verify idempotency, ensuring that running the operation again does not cause changes.
+  + After the operations, the test verifies the output files and asserts that the changes occurred as expected or not changed when re-run in check_mode, confirming the module's correct behavior and idempotency.  
+
+This structured approach ensures thorough testing of the modules' functionality and stability in different modes and repeated executions.
+
+
+5. We begin by adding tasks to post configuration to the APIC. This includes creation of all the classes such as tenant and l3out that were used in the construct_url function in our module.
+
+.. code-block:: yaml
+
+      - name: Add a new tenant
+        aci_tenant:
+          <<: *aci_info 
+          tenant: ansible_tenant
+          description: Ansible tenant
+          state: present
+
+      - name: Add a new parent object
+        aci_l3out:
+          <<: *aci_info
+          tenant: ansible_tenant
+          name: '<parent_object_name>' # Replace <parent_object_name> with the actual name of the parent object, e.g., ansible_l3out
+          description: Parent object description
+          state: present
+
+
+.. code-block:: text
+
+6. The next section consists of adding tasks for all aspects of our module. We include Ansible's register attribute to save the result of the task. The procedure is as follows:
+
+* Tasks to clean the environment to begin with a clean slate.
```

Comment:
should this not be at the start of testing, so before you create all the object required?